src/db.rs

```diff
@@ -226,7 +229,31 @@ impl DbInner {
         Ok(())
     }
 
-    pub(crate) async fn maybe_apply_backpressure(&self) {
+    #[inline]
+    pub(crate) async fn maybe_apply_wal_backpressure(&self) {
+        loop {
+            let table = {
+                let guard = self.state.read();
+                let state = guard.state();
+                if state.imm_wal.len() <= 4 {
```

Comment:
ditto: we can probably just use the same config from `self.options.max_unflushed_memtable` instead of creating a new one